drivers/led_strip/ws2812_uart.c

```diff
@@ -0,0 +1,317 @@
+/*
+ * Copyright (c) 2024 Croxel, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#define DT_DRV_COMPAT worldsemi_ws2812_uart
+
+#include <zephyr/drivers/led_strip.h>
+
+#include <string.h>
+
+#define LOG_LEVEL CONFIG_LED_STRIP_LOG_LEVEL
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(ws2812_uart);
+
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/uart.h>
+#include <zephyr/sys/math_extras.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/dt-bindings/led/led.h>
+
+struct ws2812_uart_cfg {
+	const struct device *uart_dev; /* UART device */
+	uint8_t *px_buf;
+	size_t length;
+	uint8_t num_colors;
+	const uint8_t *color_mapping;
+	const uint8_t *uart_lookup_table;
+	uint16_t reset_delay;
+	uint8_t repeat_count;
+};
+
+/*
+ * Serialize a 24-bit color channel value into an equivalent sequence
+ * of 8 bytes of UART frames.
+ */
+static inline size_t ws2812_uart_serialize(const struct device *dev, uint8_t *buf, uint8_t color[3])
+{
+	size_t buff_indx = 0;
+	uint8_t serialized_val = 0;
+	const struct ws2812_uart_cfg *cfg = dev->config;
+
+	/* Loop through each of the 3 color components (Red, Green, Blue) */
+	for (uint8_t i = 0; i < 3; i++) {
```

Comment:
3 is not correct, someone might define 1 colour, 3 colours or even 8, this needs to get it from dts